src/model/CollieHub.ts

```diff
@@ -52,12 +56,29 @@ export class CollieHub {
       ".git",
     );
     const hasAlreadyCloned = await this.git.isRepo(hubCacheGitDir);
+    let collieHubVersion = this.config.getProperty("colliehubVersion");
 
-    if (hasAlreadyCloned) {
-      await this.git.pull(hubCacheDir);
-    } else {
+    if (!hasAlreadyCloned) {
       await this.git.clone(hubCacheDir, this.url);
     }
+    if (!collieHubVersion) {
+      collieHubVersion = await this.git.getLatestTag(hubCacheDir);
+      await this.git.checkout(hubCacheDir, collieHubVersion);
+      this.config.setProperty("colliehubVersion", collieHubVersion!);
+    }
+    try {
+      const allTags = await this.git.getTags(hubCacheGitDir);
+      if (!allTags.includes(collieHubVersion)) {
+        throw new Error(
```

Comment:
d: this is strange. you throw an error here but directly catch it 3 lines later. In this case, if you dont need to stop something through multiple call-layers I suggest to just directly stop the execution in here by directly calling Deno.exit() instead of throwing an error.